packages/go/analysis/ad/esc10.go

```diff
@@ -18,109 +18,116 @@ package ad
-				if userNodes, err := ops.FetchNodeSet(tx.Nodes().Filterf(func() graph.Criteria {
-					return query.And(
-						query.KindIn(query.Node(), ad.User),
-						query.InIDs(query.NodeID(), cardinality.DuplexToGraphIDs(victimBitmap)...),
-					)
-				})); err != nil && !graph.IsErrNotFound(err) {
-					log.Warnf("Error getting user nodes for esc10a attacker nodes: %v", err)
-					continue
-				} else if len(userNodes) > 0 {
-					if subjRequireDns, err := template.Properties.Get(ad.SubjectAltRequireDNS.String()).Bool(); err != nil {
-						log.Debugf("Failed to retrieve subjectAltRequireDNS for template %d: %v", template.ID, err)
-						victimBitmap.Xor(cardinality.NodeSetToDuplex(userNodes))
-					} else if subjRequireDomainDns, err := template.Properties.Get(ad.SubjectAltRequireDomainDNS.String()).Bool(); err != nil {
-						log.Debugf("Failed to retrieve subjectAltRequireDomainDNS for template %d: %v", template.ID, err)
-						victimBitmap.Xor(cardinality.NodeSetToDuplex(userNodes))
-					} else if subjRequireDns || subjRequireDomainDns {
-						//If either of these properties is true, we need to remove all these users from our victims list
-						victimBitmap.Xor(cardinality.NodeSetToDuplex(userNodes))
-					}
-				}
+				victimBitmap := getVictimBitmap(groupExpansions, certTemplateControllers, ecaControllers)
 
-				if attackers, err := ops.FetchStartNodes(tx.Relationships().Filterf(func() graph.Criteria {
-					return query.And(
-						query.KindIn(query.Start(), ad.Group, ad.User, ad.Computer),
-						query.KindIn(query.Relationship(), ad.GenericAll, ad.GenericWrite, ad.Owns, ad.WriteOwner, ad.WriteDACL),
-						query.InIDs(query.EndID(), cardinality.DuplexToGraphIDs(victimBitmap)...),
-					)
-				})); err != nil {
+				if filteredVictims, err := filterUserDNSResults(tx, victimBitmap, template); err != nil {
+					log.Warnf("error filtering users from victims for esc9a: %v", err)
+					continue
+				} else if attackers, err := FetchAttackersForEscalations9and10(tx, filteredVictims, false); err != nil {
 					log.Warnf("Error getting start nodes for esc10a attacker nodes: %v", err)
 					continue
 				} else {
-					results.Or(cardinality.NodeSetToDuplex(attackers))
+					results.Or(cardinality.NodeIDsToDuplex(attackers))
 				}
 			}
 		}
 
 		results.Each(func(value uint32) bool {
-			if !channels.Submit(ctx, outC, analysis.CreatePostRelationshipJob{
+			channels.Submit(ctx, outC, analysis.CreatePostRelationshipJob{
 				FromID: graph.ID(value),
 				ToID:   domain.ID,
 				Kind:   ad.ADCSESC10a,
-			}) {
-				return false
-			} else {
-				return true
-			}
+			})
+			return true
 		})
+	}
+	return nil
+}
 
+func PostADCSESC10b(ctx context.Context, tx graph.Transaction, outC chan<- analysis.CreatePostRelationshipJob, groupExpansions impact.PathAggregator, enterpriseCA, domain *graph.Node, cache ADCSCache) error {
+	if canAbuseUPNRels, err := FetchCanAbuseUPNCertMappingRels(tx, enterpriseCA); err != nil {
```

Comment:
According to the post processing doc, this should be `CanAbuseWeakCertBinding`

Reply:
It seems there was a typo in the docs. Thanks for the thorough review!